runtime/graphql/schemagen/schemagen.go

```diff
@@ -128,12 +129,14 @@ type FieldDefinition struct {
 	Name       string
 	Arguments  []*ArgumentDefinition
 	ReturnType string
+	Docs       *metadata.Docs
 }
 
 type TypeDefinition struct {
 	Name      string
-	Fields    []*NameTypePair
+	Fields    []*metadata.Field
 	IsMapType bool
+	Docs      *metadata.Docs
```

Comment:
`TypeDefinition` is about the GraphQL type definition.  It shouldn't have fields or docs that are part of the metadata.  In other words, the Metadata fields are the fields within the original function, while the object here is about the generated _GraphQL_ fields.

You may need to make some new types to support this.

Also, the schemagen tests don't compile, because of this.